src/backend/distributed/commands/extension.c

```diff
@@ -115,3 +110,516 @@ ExtractNewExtensionVersion(Node *parsetree)
 
 	return newVersion;
 }
+
+
+/*
+ * PlanCreateExtensionStmt is called during the creation of an extension.
+ * It is executed before the statement is applied locally.
+ * We decide if the extension needs to be replicated to the worker, and
+ * if that is the case return a list of DDLJob's that describe how and
+ * where the extension needs to be created.
+ */
+List *
+PlanCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt, const char *queryString)
+{
+	List *commands = NIL;
+	const char *createExtensionStmtSql = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) createExtensionStmt))
+	{
+		return NIL;
+	}
+
+	/* Extension management can only be done via coordinator node */
+	EnsureCoordinator();
+
+	/*
+	 * Here we append "new_version" and "schema" fields to the "options" list
+	 * if not specified to satisfy the version and schema consistency
+	 * between worker nodes and the coordinator.
+	 */
+	AddMissingFieldsCreateExtensionStmt(createExtensionStmt);
+
+	createExtensionStmtSql = DeparseTreeNode((Node *) createExtensionStmt);
+
+	/*
+	 * To prevent recursive propagation in mx architecture, we disable ddl
+	 * propagation before sending the command to workers.
+	 */
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) createExtensionStmtSql,
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * Add "new_version" and "schema" DefElem items (if not specified in
+ * statement) to "options" list before deparsing the statement.
+ */
+static void
+AddMissingFieldsCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt)
+{
+	List *optionsList = createExtensionStmt->options;
+
+	const char *newVersion = GetCreateExtensionOption(optionsList, "new_version");
+	const char *schemaName = GetCreateExtensionOption(optionsList, "schema");
+
+	if (!newVersion)
+	{
+		char *extensionName = createExtensionStmt->extname;
+
+		/*
+		 * Get the latest available version of the extension from
+		 * pg_available_extensions() to ensure version consistency
+		 * between worker nodes and coordinator.
+		 */
+		char *extensionDefaultVersion = GetDefaultExtensionVersion(extensionName);
+
+		Node *extensionVersionArg = (Node *) (makeString(extensionDefaultVersion));
+
+		/* Set location to -1 as it is unknown */
+		int location = -1;
+
+		DefElem *newDefElement = makeDefElem("new_version", extensionVersionArg,
+											 location);
+
+		createExtensionStmt->options = lappend(createExtensionStmt->options,
+											   newDefElement);
+	}
+	if (!schemaName)
+	{
+		char *currentSchemaName = GetCurrentSchema();
+
+		Node *schemaNameArg = (Node *) (makeString(currentSchemaName));
+
+		/* Set location to -1 as it is unknown */
+		int location = -1;
+
+		DefElem *newDefElement = makeDefElem("schema", schemaNameArg, location);
+
```

Comment:
Let's add a comment here. Why are we adding `schema` here?

Reply:
I guess I should extend the leading comment of this function